README.MD

```diff
@@ -0,0 +1,326 @@
+
+2. GatewayComposer: A Gateway Composer facilitates staking value for branded tokens and minting a utility representation of those branded tokens with a Utility Branded Token through a gateway for use within the given application, thereby reducing both the number of transactions to execute and the number of contracts to call in order to stake and mint.
+
+3. UtilityBrandedToken: A Utility Branded Token, which also implements the EIP-20 Standard Token interface, is the utility token representation of the Branded Token on a sidechain. This representation is orchestrated via a gateway, comprising a Gateway contract on the same chain as the Branded Token, and a CoGateway contract on the sidechain with the Utility Branded Token.
+
+The following steps assume that nodejs and geth are installed and running.
+This code was tested with geth version: 1.7.3-stable. Other higher versions should also work.
+
+#### Basic setup
+
+1.  Install BrandedToken.js in your project using npm
+
+```bash
+$ npm install @openstfoundation/brandedtoken.js --save
+
+```
+
+2.  Sync your development machine with one of the following test environments
+
+    - https://ropsten.etherscan.io/
+    - https://kovan.etherscan.io/
+    - https://rinkeby.etherscan.io/
+
+3.  Create the following addresses and fund them with gas
+
+    - deployerAddress - address that deploys the EIP20Token, BrandedToken, UtilityBrandedToken and GatewayComposer contracts.
+    - facilitator - address that calls acceptStakeRequest and rejectStakeRequest.
+    - owner - address that calls GatewayComposer.requestStake.
+    - worker - address that calls BrandedToken.rejectStakeRequest.
+
+
+
+#### Creating a BrandedToken object
+The Branded Token object is an entry point for developer - using the BrandedToken object, a staking can be initiated.
+
+```js
+// Creating brandedtoken.js object
+const BrandedToken = require('./index.js');
+const Setup = BrandedToken.EconomySetup;
+
+```
+
+######  Constants
+
+Please can set below constants to addresses that you are able to control.
+
+```js
+
+// Initialize web3 object using the geth endpoint
+const web3Provider = new Web3('http://127.0.0.1:8545');
+
+// organization owner. Doesn't need to be eth funded.
+let organizationOwner = '0xaabb1122....................';
+
+// deployer address
+let deployerAddress = '0xaabb1122....................';
+
+// Facilitator address
+let facilitator = '0xaabb1122....................';
+
+// staker address
+let staker = '0xaabb1122....................';
+
+// worker address
+let worker = '0xaabb1122....................';
+
+let passphrase = 'some passphrase.....';
+
+// Other constants
+const gasPrice = '0x12A05F200';
+const gas = 7500000;
+
+```
+
+###### ABI and BIN provider
+
+brandedtoken.js comes with an in-built abi-bin provider for managing abi(s) and bin (s).
+
+The abiBinProvider provides developers with following abi(s) and bin(s):
```

Comment:
Suggest: "The abiBinProvider provides abi(s) and bin(s) for the following contracts:"